ProcessLib/LIE/HydroMechanics/HydroMechanicsProcess.cpp

```diff
@@ -568,13 +574,17 @@ void HydroMechanicsProcess<GlobalDim>::assembleWithJacobianConcreteProcess(
 {
     DBUG("AssembleWithJacobian HydroMechanicsProcess.");
 
+    const int process_id =
+        _use_monolithic_scheme ? 0 : _coupled_solutions->process_id;
+    ProcessLib::ProcessVariable const& pv = getProcessVariables(process_id)[0];
```

Comment:
For another PR; Maybe also a `processVariable(int)` function too with appropriate safety checks.

Reply:
TODO once this PR is merged too.